swift/ql/lib/codeql/swift/elements/decl/IterableDeclContext.qll

```diff
@@ -1,4 +1,26 @@
-// generated by codegen/codegen.py, remove this comment if you wish to edit this file
 private import codeql.swift.generated.decl.IterableDeclContext
+private import codeql.swift.elements.decl.Decl
+private import codeql.swift.elements.decl.VarDecl
+private import codeql.swift.elements.decl.SubscriptDecl
 
-class IterableDeclContext extends Generated::IterableDeclContext { }
+class IterableDeclContext extends Generated::IterableDeclContext {
+  /**
+   * Gets the `index`th member of this iterable declaration context (0-based),
+   * including `AccessorDecl`s of immediate members.
+   */
+  override Decl getImmediateMember(int index) {
+    result =
+      rank[1 + index](Decl member, Decl immMember, int immIndex, int accIndex |
+        immMember = super.getImmediateMember(immIndex) and
+        (
+          member = immMember and accIndex = -1
+          or
+          member = immMember.(VarDecl).getImmediateAccessorDecl(accIndex)
+          or
+          member = immMember.(SubscriptDecl).getImmediateAccessorDecl(accIndex)
+        )
+      |
+        member order by immIndex, accIndex
+      )
+  }
+}
```

Comment:
I'm not really sold on this change, as I do think accessors are logically and visually under a `VarDecl` or `SubscriptDecl` (and by visually I mean they are declared in subblocks of variable/subscript declarations). I would rather keep them that way.

Even if we do go with this, this requires taking them out of the children of `VarDecl`/`SubscriptDecl`, as the `ast/no_double_parents.ql` test failure shows.